keps/sig-storage/4958-csi-sidecars-all-in-one/README.md

```diff
@@ -0,0 +1,1214 @@
+
+
+#### Milestone-Build-the-project-using-a-modified-copy-of-release-tools
+
+
+#### Milestone-set-up-new-test-infra-jobs-to-test-the-project-through-the-hostpath-CSI-Driver
+
+
+#### Milestone-migration-of-CSI-Drivers-to-the-new-model 
+
+
+---
+
+
+#### Milestone-all-six-sidecar-repo-had-been-integrated-into-monorepo
+
+Alpha phase
+
+#### Milestone-be-ready-to-accept-PR-from-community
+
+Beta phase
+
+#### Milestone-six-sidecars-working-through-CSI-hostpath
+
+Beta phase
+
+#### Milestone-three-cloud-vendors-start-using-the-monorepo-component
+
+GA phase
+
+#### Milestone-all-individual-repo-has-been-into-deprecated-state
+
+Standalone phase
+
+#### Milestone-merge-sidecar-informer-caches
+
+Standalone phase
+
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+### Graduation Criteria
```

Comment:
We should have already worked in the workstream about CSI Driver migration

- Beta graduation would be 2 CSI Drivers using the AIO sidecar for at least 2 consecutive k8s minor releases.
- Stable graduation would be 2 CSI Drivers using the AIO sidecar for 3 consecutive k8s minor releases.